ramp/core/core.py

```diff
@@ -60,30 +61,147 @@ def __init__(self, user, number = 1, power = 0, num_windows = 1, func_time = 0,
         self.wd_we = wd_we_type #defines if the App is associated with weekdays or weekends | 0 is wd 1 is we 2 is all week
-        if P_series == False and isinstance(power, pd.DataFrame) == False: #check if the user defined P as timeseries
+        if P_series == False and not isinstance(power, pd.DataFrame): #check if the user defined P as timeseries
             self.POWER = power*np.ones(365) #treat the power as single value for the entire year
         else:
             self.POWER = power.values[:,0] #if a timeseries is given the power is treated as so
 
     def windows(self, w1 = np.array([0,0]), w2 = np.array([0,0]),r_w = 0, w3 = np.array([0,0])):
-        self.window_1 = w1 #array of start and ending time for window of use #1
-        self.window_2 = w2 #array of start and ending time for window of use #2
-        self.window_3 = w3 #array of start and ending time for window of use #3
-        self.random_var_w = r_w #percentage of variability in the start and ending times of the windows
-        self.daily_use = np.zeros(1440) #create an empty daily use profile
-        self.daily_use[w1[0]:(w1[1])] = np.full(np.diff(w1),0.001) #fills the daily use profile with infinitesimal values that are just used to identify the functioning windows
-        self.daily_use[w2[0]:(w2[1])] = np.full(np.diff(w2),0.001) #same as above for window2
-        self.daily_use[w3[0]:(w3[1])] = np.full(np.diff(w3),0.001) #same as above for window3
+        self.random_var_w = r_w  # percentage of variability in the start and ending times of the windows
+        self.daily_use = np.zeros(1440)  # create an empty daily use profile
+        for count, window_x in enumerate((w1,w2,w3),start=1):
+            self.__setattr__(f'window_{count}',window_x) #array of start and ending time for window of use #1,2 & 3
+            self.daily_use[window_x[0]:window_x[1]] = np.full(np.diff(window_x), 0.001) #fills the daily use profile with infinitesimal values that are just used to identify the functioning windows
+            self.__setattr__(f'random_var_{count}', int(r_w*np.diff(window_x))) #calculate the random variability of the windows, i.e. the maximum range of time they can be enlarged or shortened
         self.daily_use_masked = np.zeros_like(ma.masked_not_equal(self.daily_use,0.001)) #apply a python mask to the daily_use array to make only functioning windows 'visibile'
-        self.random_var_1 = int(r_w*np.diff(w1)) #calculate the random variability of window1, i.e. the maximum range of time they can be enlarged or shortened
-        self.random_var_2 = int(r_w*np.diff(w2)) #same as above
-        self.random_var_3 = int(r_w*np.diff(w3)) #same as above
         self.user.App_list.append(self) #automatically appends the appliance to the user's appliance list
 
         if self.activate == 1:
             self.cw11 = self.window_1
             self.cw12 = self.window_2
 
+    def calc_rand_window(self, window_num=1, max_window_range=np.array([0, 1440])):
```

Comment:
I would add a docstring (""" write docstring description""", below the method definition) to explain what the method is doing